src/ansible_navigator/configuration_subsystem/configurator.py

```diff
@@ -234,7 +234,17 @@ def _apply_cli_params(self) -> None:
                 self._messages.append(LogMessage(level=logging.INFO, message=message))
 
     def _post_process(self) -> None:
+        delayed = []
+        normal = []
+
+        # Sort delayed and normal entries.
         for entry in self._config.entries:
+            if entry.delay_post_process:
+                delayed.append(entry)
+            else:
+                normal.append(entry)
+
+        for entry in normal + delayed:
```

Comment:
I don't know if the memory consumption is important here but if it is, maybe make this lazy:
```suggestion
        for entry in itertools.chain(normal, delayed):
```

Reply:
I hope we never have _that_ many :)   If we did, making 2 passes through the entries first for normal, second for delayed would avoid the creation of the new lists all together. Something like this:

```
for should_delay in [False, True]:
    if entry.delay_post_process is should_delay:
        if self._initial or entry.change_after_initial:
            processor = getattr(self._config.post_processor, entry.name, None)
            if callable(processor):
                messages, errors = processor(entry=entry, config=self._config)
                self._messages.extend(messages)
                self._exit_messages.extend(errors)
        else:
            message = f"'{entry.name}' cannot be reconfigured. (post process)"
            self._messages.append(LogMessage(level=logging.INFO, message=message))
```
that said, I think it's fine the way it is.

Reply:
(untested of course)

Reply:
> If we did, making 2 passes through the entries first for normal, second for delayed would avoid the creation of the new lists all together.

Actually, if we need this sort of optimization, knowing that we operate on a list, I'd probably not even create these two extra lists in the first place but have a sequence of IDs pointing to one or another. Or, even better, I'd combine it with separating this into a generator function for clarity.